core/src/main/java/com/alibaba/nacos/core/code/ControllerMethodsCache.java

```diff
@@ -116,34 +173,52 @@ private void parseSubAnnotations(Method method, String classPath) {
         final PatchMapping patchMapping = method.getAnnotation(PatchMapping.class);
         
         if (getMapping != null) {
-            put(RequestMethod.GET, classPath, getMapping.value(), method);
+            put(RequestMethod.GET, classPath, getMapping.value(), getMapping.params(), method);
         }
         
         if (postMapping != null) {
-            put(RequestMethod.POST, classPath, postMapping.value(), method);
+            put(RequestMethod.POST, classPath, postMapping.value(), postMapping.params(), method);
         }
         
         if (putMapping != null) {
-            put(RequestMethod.PUT, classPath, putMapping.value(), method);
+            put(RequestMethod.PUT, classPath, putMapping.value(), putMapping.params(), method);
         }
         
         if (deleteMapping != null) {
-            put(RequestMethod.DELETE, classPath, deleteMapping.value(), method);
+            put(RequestMethod.DELETE, classPath, deleteMapping.value(), deleteMapping.params(),
+                    method);
         }
         
         if (patchMapping != null) {
-            put(RequestMethod.PATCH, classPath, patchMapping.value(), method);
+            put(RequestMethod.PATCH, classPath, patchMapping.value(), patchMapping.params(),
+                    method);
         }
         
     }
-    
-    private void put(RequestMethod requestMethod, String classPath, String[] requestPaths, Method method) {
+
+    private void put(RequestMethod requestMethod, String classPath, String[] requestPaths,
+            String[] requestParams, Method method) {
         if (ArrayUtils.isEmpty(requestPaths)) {
-            methods.put(requestMethod.name() + "-->" + classPath, method);
+            String urlKey = requestMethod.name() + "-->" + classPath;
+            addUrlAndMethodRelation(urlKey, requestParams, method);
             return;
         }
         for (String requestPath : requestPaths) {
-            methods.put(requestMethod.name() + "-->" + classPath + requestPath, method);
+            String urlKey = requestMethod.name() + "-->" + classPath + requestPath;
```

Comment:
The symbol "-->" appears multiple times and is recommended as a static fixed variable.

Reply:
it looks better